README.md

```diff
@@ -111,6 +111,129 @@ The custom middleware can then be configured like the following:
 
+## Profile Server
+
+This option allows for profiles to be passively collected via an HTTP endpoint,
+inspired by [golang's built-in pprof server](https://pkg.go.dev/net/http/pprof).
+
+A minimal Rack app runs a minimal (non-compliant) HTTP server, which exposes an
+endpoint that allows for profiling. For security purposes, the server is bound
+to localhost only. The HTTP server is built using standard library modules only,
+in order to keep dependencies minimal. Because it is an entirely separate server,
+listening on an entirely separate socket, this should not interfere with any
+existing application routes, and should even be usable in non-web apps.
+
+This allows for two main use cases:
+
+- Passive profile collection in production
+  - Periodically profiling production apps to analyze them responding to real
+workloads
+  - Providing a statistical, long-term view of the "hot paths" of the workload
+- Local development profiling
+  - Can be used to get a profile "on demand" against a development server
+
+### Configuration
+
+If using as a railtie, only a single option needs to be set:
+
+```
+config.app_profiler.server_enabled = true
+```
+
+Alternatively, the server can be directly started with:
+
+```
+AppProfiler::Server.start!
+```
+
+The default duration, for requests without a duration parameter, can also be
+set via the railtie config.
+
+```
+AppProfiler.server.duration
+```
+
+The server supports both TCP and Unix sockets for its transport. It is
+recommended to use TCP for local development, and Unix sockets for production:
+
+```
+AppProfiler.server.transport = AppProfiler::Server::TRANSPORT_UNIX
+```
+
+It is possible, but not recommended, to hardcode the listen port to be used in
+TCP server mode with:
+
+```
+AppProfiler.server.port
+```
+
+If this is done in production and it can cause port conflicts with multiple
+instances of the app, which is another reason why the Unix transport is
+preferred for production.
+
+#### Discovering the port or socket path
+
+In general, the server should be run without setting the port, in which case
+any free TCP port may be used. To determine what the port is, check the
+application logs, or resolve it from the special "Magic file" which contains
+a mapping of pid to port:
+
+```
+$ PID=49825
+$ port_file=$(ls -1 /tmp/app_profiler/profileserver-$PID-port-*)
+$ PORT=$(echo $port_file | sed 's/.*port-\([[:digit:]]*\)-.*/\1/g')
+$ echo $PORT
+60160
+```
+
+This approach is intended to be "machine friendly" so that an external
+profiling agent can easily detect what port to profile on.
+
+For the Unix mode, this is even easier as the file simply includes the PID
+in it, and this will be the file handle to use for the Unix socket:
+
+```
+$ PID=41016
+$ SOCK=$(ls -1d /tmp/app_profiler/* | grep $PID.sock)
+$ echo $SOCK
+/tmp/app_profiler/app-profiler-41016.sock
+```
+
+### Collecting a profile
+
+The API is very simple, and passes supported parameters directly to stackprof.
+
+Supported are:
```

Comment:
Can we just link to the section above this that explains the possible options instead of reiterating them?

Reply:
Good idea 👍